Cal poly ratings support

mfish33 opened the pull request:
Many professors are not found on polyratings. In order to combat this I added a call to calpolyratings if the professor is not found on polyratings. This adds ratings to a large amount of professors and does not change any older features.

Commit:
Cal poly ratings support

Commit:
Cal poly ratings support

Commit:
Merge branch 'master' of https://github.com/mfish33/PassThePlebs

Commit:
fixed accidental misspelling

Commit:
fixed regex

Commit:
removed redundent class var declaration

timstoddard commented on src/extension/js/add-polyratings.js:
```diff
@@ -1,40 +1,32 @@
-  showBackgroundColors;
-  staffClassesOption;
+// Maps calpolyratings to polyratings scale for color picker
+const CALPOLYRATINGSMAP = {
```
add underscores between words: `CAL_POLY_RATINGS_MAP `

timstoddard commented on src/extension/js/add-polyratings.js:
```diff
@@ -107,7 +96,7 @@ export default class PolyratingIntegrator {
       })
       const fullName = this.urlFormat(`${firstNames} ${lastName}`)
-      if (namesList.indexOf(fullName) === -1) {
+      if (!namesList.includes(fullName)) {
```
nice

timstoddard commented on src/extension/js/add-polyratings.js:
```diff
@@ -205,6 +189,34 @@ export default class PolyratingIntegrator {
 
+  checkCalPolyRatings(rawname, nameElems) {
+    const [lastName, firstName] = rawname.split(/[, ]/)
+      .filter(name => name.length > 2 && name.split('I').length !== name.length + 1) // detects II and M.
```
whats the purpose of detecting "II and M"?

mfish33 replied:
II is the titles like the first second and third. I also saw a teacher with m. before there name. I am doing this try and get more accurate first and last names since I don't iterate through names like your solution.

timstoddard commented on src/extension/js/add-polyratings.js:
```diff
@@ -205,6 +189,34 @@ export default class PolyratingIntegrator {
+    const proxyUrl = 'https://cors-anywhere.herokuapp.com/'
+    const herf = `https://www.calpolyratings.com/${firstName.toLowerCase()}-${lastName.toLowerCase()}/`
+
+    fetch(proxyUrl + herf).then(res => res.text()).then(data => {
```
Please use the chrome extension api for making http calls, as is done earlier in this file.

```
chrome.runtime.sendMessage({
  method: 'GET',
  action: 'xhttp',
  url: 'YOUR_URL',
  data: '',
},
response => {
  // do stuff here
})
```

timstoddard commented on src/extension/js/add-polyratings.js:
```diff
@@ -205,6 +189,34 @@ export default class PolyratingIntegrator {
+
+    fetch(proxyUrl + herf).then(res => res.text()).then(data => {
+      // Generate random string for dom element to avoid conflict in parsing
+      const domID = (new Array(10)).fill(null).map(_ => String.fromCharCode((Math.ceil(Math.random() * 26) + 96))).join('')
```
Creating a random ID and a new element is unnecessary here. Once you have the response from calpolyratings, the workflow should just be to grab the ratings/eval from the returned html using css selectors to find the specific elements you want. It should look something like how I did it for the other site:
```
const ratingElem = polyratingPage.find('.hidden-xs > span > .text-primary')
if (ratingElem[0]) {
  const rating = ratingElem.text()
  const profId = response.match(/profid=(\d+)/)[1]
  const evals = ratingElem.next().text().replace('uation', '')
  const href = `http://polyratings.com/eval.php?profid=${profId}`
  const numericalRating = parseFloat(rating, 10)
  const bgColor = this.calculateBackgroundColor(numericalRating)
  this.updateInstructorName(rawName, nameElems, bgColor, href, rating, evals)
} else {
  // not found
}
```

timstoddard commented on src/extension/js/add-polyratings.js:
```diff
@@ -217,7 +229,7 @@ export default class PolyratingIntegrator {
   }
 
-  /*** UTILS ***/
+  /** * UTILS ** */
```
nit: remove spaces

timstoddard reviewed (changes requested):
You've got a good start here. Thanks for taking the time to work on this. Let me know if you have questions about my feedback.

mfish33 committed:
fixed build tooling to support windows

mfish33 committed:
requested changes

mfish33 committed:
spelling fix

mfish33 committed:
spelling fix

mfish33 committed:
improved build tooling to work acrross platforms

mfish33 commented:
I changed copying files to be handled with webpack. Also, I added a build file to produce the final zip. This is because the only post-install webpack zipping plugin uses the zip command under the hood and causes errors on windows. Now you can run npm run build without any errors across platforms.

timstoddard commented on package.json:
```diff
@@ -47,6 +48,8 @@
     "webpack": "4.28.3",
     "webpack-cli": "3.2.0",
-    "webpack-merge": "4.2.1"
+    "archiver": "^3.1.1",
```
deps should be alphabetized (this will happen automatically if you install from the command line via `npm i -D <package_name>`)

mfish33 replied:
oops. I accidentally installed them as regular deps and moved them manually wich I forgot to alphabetize.

timstoddard commented on src/extension/js/add-polyratings.js:
```diff
@@ -191,31 +191,33 @@ export default class PolyratingIntegrator {
+    .filter(name => name.length > 2 && name.split('I').length !== name.length + 1) // detects II and M.
 
     const proxyUrl = 'https://cors-anywhere.herokuapp.com/'
     const herf = `https://www.calpolyratings.com/${firstName.toLowerCase()}-${lastName.toLowerCase()}/`
```
fix spelling (href)

timstoddard reviewed (changes requested):
Thanks for making the changes I asked for. There's just 2 very minor fixes left to make. Nice work on the new webpack stuff, I've only worked on this on my mac but it's always good to be platform agnostic :)

mfish33 committed:
fixed spelling and deps

mfish33 commented:
resolves https://github.com/timstoddard/PassThePlebs/issues/3

mfish33 committed:
fixed variable name

timstoddard commented:
Thanks again for the PR. I'm gonna do some more testing and push out a new version with your changes soon!

timstoddard commented:
FYI @mfish33, just published a new version (1.2.0) that includes your changes (might take a little while to propagate to the extension store). I'll need to do some more testing once PASS is live again (2/6), so let me know if you run into any bugs with the new code.
